### PR TITLE
Support for CentOS 8

### PR DESCRIPTION
**Pull request by akutz**
This patch introduces initial support for building OVAs with CentOS 8.

Hi @codenrhoden @figo @ppadmavilasom,

Please see the following comment in the kickstart file:

> There was an attempt in 2037e4f3b0e272883c64102ca47f175fc8faf71a to no longer install packages from the internet during the kickstart of CentOS 7.
>
> However, CentOS 8 no longer provides a minimal ISO, only the full DVD or the network boot ISOs. This means either a +7GiB DVD ISO is required to build  CentOS 8, or a 500MiB-ish ISO that requires network access may be used.
>
> The initial work to support CentOS 8 has opted to use the network boot ISO as the ESX builder fails when using large ISOs as learned while attempting to support Photon 3 Rev2.

This patch will be rebased once #121 is merged.

### CODE REVIEW

**Comment by akutz** [comment]
The Ansible roles need to be updated to use the EPEL 8 repo when using CentOS 8, otherwise errors like the following occur:

```shell
 centos-8: TASK [common : perform a yum update] *******************************************
    centos-8: fatal: [default]: FAILED! => {"changed": false, "failures": [], "msg": "Depsolve Error occured: \n Problem: cannot install the best update candidate for package libidn2-2.0.5-1.el8.x86_64\n  - nothing provides libunistring.so.0()(64bit) needed by libidn2-2.3.0-1.el7.x86_64", "rc": 1, "results": []}
```

**Comment by akutz** [comment]
/hold cancel

Should be good to go!

```shell
$ cat /etc/redhat-release 
CentOS Linux release 8.0.1905 (Core) 
```

```shell
$ sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf get all -A
NAMESPACE     NAME                                  READY   STATUS    RESTARTS   AGE
kube-system   pod/coredns-5644d7b6d9-qh7xm          0/1     Pending   0          82s
kube-system   pod/coredns-5644d7b6d9-wksr6          0/1     Pending   0          82s
kube-system   pod/etcd-capi.vm                      1/1     Running   0          29s
kube-system   pod/kube-apiserver-capi.vm            1/1     Running   0          24s
kube-system   pod/kube-controller-manager-capi.vm   1/1     Running   0          18s
kube-system   pod/kube-proxy-hhgxf                  1/1     Running   0          82s
kube-system   pod/kube-scheduler-capi.vm            1/1     Running   0          32s

NAMESPACE     NAME                 TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)                  AGE
default       service/kubernetes   ClusterIP   100.64.0.1    <none>        443/TCP                  99s
kube-system   service/kube-dns     ClusterIP   100.64.0.10   <none>        53/UDP,53/TCP,9153/TCP   98s

NAMESPACE     NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                 AGE
kube-system   daemonset.apps/kube-proxy   1         1         1       1            1           beta.kubernetes.io/os=linux   98s

NAMESPACE     NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
kube-system   deployment.apps/coredns   0/2     2            0           98s

NAMESPACE     NAME                                 DESIRED   CURRENT   READY   AGE
kube-system   replicaset.apps/coredns-5644d7b6d9   2         2         0       82s
```

**Comment by akutz** [comment]
FYI, the resulting OVA is around 1.2GiB:

```shell
$ ls -al output/centos-8-kube-v1.16.2/*.ova
-rw-r--r--  1 akutz  staff   1.2G Jan 12 21:34 output/centos-8-kube-v1.16.2/centos-8.ova
```

File: images/capi/ansible/roles/common/defaults/main.yml [hunk 1]
```diff
@@ -18,13 +18,19 @@ common_rpms:
...
+- yum-utils
+common_7_rpms:
+- ntp
 - python2-pip
 - python-netifaces
 - python-requests
-- socat
-- yum-utils
+common_8_rpms:
+- chrony
```
Comment (codenrhoden) [inline images/capi/ansible/roles/common/defaults/main.yml]: Just for posterity, I think this brings up the topic of https://github.com/kubernetes-sigs/image-builder/issues/20 again. I don't think it should affect this PR, but it's still an open question if we want to standardize on NTPD or not.

**Comment by codenrhoden** [comment]
/remove-lifecycle stale
This will likely get revisited in the next couple of months

**Comment by codenrhoden** [comment]
/lifecycle frozen

I don't want this to close. It's still important and on the roadmap! I will revisit this once CI is in place for the existing OS and provider combos.

**Comment by codenrhoden** [comment]
I'm going to go ahead and close this one for now, as we'll need to re-evaluate our approach with the move to CentOS Stream. This still may serve as a very good guide for implementation in the future.

/close

**Pull request closed without merging**